cst438_register/src/test/java/com/cst438/StudentControllerTest.java

```diff
@@ -0,0 +1,148 @@
+package com.cst438;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import com.cst438.controller.StudentController;
+import com.cst438.domain.Enrollment;
+import com.cst438.domain.ScheduleDTO;
+import com.cst438.domain.Student;
+import com.cst438.domain.StudentDTO;
+import com.cst438.domain.StudentRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@ContextConfiguration(classes = { StudentController.class })
+@AutoConfigureMockMvc(addFilters = false)
+@WebMvcTest
+class StudentControllerTest {
+	public static final String Test_Student_Name = "Jonathan Floor" ;
+	public static final String Test_Student_Email = "JFloor@csumb.edu";
+	public static final String Test_Present_Student_Email = "test@csumb.edu";
+	public static final String Test_Present_Student_Name = "test";
+
+
+	@MockBean
+	StudentRepository studentRepository;
+	
+
+	@Autowired
+	private MockMvc mvc;
+	
+	@Test
+	public void createNewStudent() throws Exception {
+		MockHttpServletResponse response;
+		Student Newstudent = new Student();
+		Newstudent.setEmail(Test_Student_Email);
+		Newstudent.setEmail(Test_Student_Email);
+		
+		given(studentRepository.save(any(Student.class))).willReturn(Newstudent);
+		
+	    StudentDTO studentDTO = new StudentDTO();
+	    studentDTO.email = Test_Student_Email;
+	    studentDTO.name = Test_Student_Name;
+	    
+	    
+		response = mvc.perform(
+				MockMvcRequestBuilders
+			      .post("/student")
+			      .content(asJsonString(studentDTO))
+			      .contentType(MediaType.APPLICATION_JSON)
+			      .accept(MediaType.APPLICATION_JSON))
+				.andReturn().getResponse();
+		
```

Comment:
Delete some of this extra empty space to help make this section a little cleaner.